src/main/java/de/csbdresden/stardist/StarDist2D.java

```diff
@@ -400,4 +407,50 @@ public static void main(final String... args) throws Exception {
         ij.command().run(StarDist2D.class, true, params);
     }
 
+    private void refresh() {
+        final HashMap<String, Object> params = new HashMap<>();
+
+        params.put("input", input);
+        params.put("modelChoice", modelChoice);
+        params.put("normalizeInput", true);
+        params.put("percentileBottom", percentileBottom);
+        params.put("percentileTop", percentileTop);
+        params.put("probThresh", probThresh);
+        params.put("nmsThresh", nmsThresh);
+        params.put("outputType", "ROI Manager");
+        params.put("nTiles", 1);
+        params.put("excludeBoundary", 2);
+        params.put("roiPosition", Opt.ROI_POSITION_AUTO);
+        params.put("preview", false);
+
+        command.run(StarDist2D.class, false, params);
+    }
+
+    @Override
+    public void preview() {
+        if(preview) {
+            refresh();
+        } else {
+            cancel();
+        }
+    }
+
+    @Override
+    public void cancel() {
+        RoiManager.getInstance().reset();
```

Comment:
I was getting a null pointer exception here. But I did work when I replaced this line with [this](https://github.com/stardist/stardist-imagej/blob/6aff9ed952896dce954cda08602add5ee3425c5f/src/main/java/de/csbdresden/stardist/StarDist2DBase.java#L94-L98).